### PR TITLE
rapiers

### PR DESCRIPTION
**Pull request by RealestEstate**
introduces rapiers
they have 2 stances:
-a high stance which does 20 force
-a low stance which has a chance to parry but only does 13 force

they're essentially shittier katanas
viedo =
https://streamable.com/xuxoy

:cl:
 * rscadd: Adds rapiers, they can be toggled to switch between a power stance and a parrying stance.

### CODE REVIEW

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -475,3 +475,64 @@ obj/item/weapon/banhammer/admin
...
+	else
+		set_high_stance()
+	usr.update_inv_hands()
+	playsound(src, 'sound/weapons/swish.ogg', 25, 1)
+
+
+/obj/item/weapon/rapier/attack_self(mob/living/user as mob)
+	if(cooldown < world.time - 2)
+		swap_stance()
+		user.visible_message("<span class='warning'>You switch your stance with the [src] to [stance]!</span>")
```
Comment (unid15) [inline code/game/objects/items/weapons/weaponry.dm]: This message will be displayed to everybody around

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -475,3 +475,64 @@ obj/item/weapon/banhammer/admin
...
+		cooldown = world.time
+	else
+		..()
+
+/obj/item/weapon/rapier/suicide_act(mob/user)
+	to_chat(viewers(user), "<span class='danger'>[user] is impaling \himself with the [src.name]! It looks like \he's trying to commit suicide.</span>")
```
Comment (unid15) [inline code/game/objects/items/weapons/weaponry.dm]: Is there a reason not to use visible_message in suicide_act, other than old habits?

**Review by unid15 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (RealestEstate) [reply code/game/objects/items/weapons/weaponry.dm]: how do I make it only display itself to the user?

Comment (unid15) [reply code/game/objects/items/weapons/weaponry.dm]: user.visible_message("[user] does a thing", "You do a thing")

**Comment by unid15** [comment]
aw are there no sprite difference for the stances? that'd be cool

File: code/game/objects/items/weapons/weaponry.dm (hunk 2 above)
Comment (DamianX) [reply code/game/objects/items/weapons/weaponry.dm]: visible_message is definitely the way to go

**Comment by DamianX** [comment]
Spessmart-only?

File: code/game/objects/items/weapons/weaponry.dm [hunk 3]
```diff
@@ -475,3 +475,64 @@ obj/item/weapon/banhammer/admin
...
+	throwforce = 5
+	w_class = W_CLASS_MEDIUM
+	sharpness = 1.2
+	sharpness_flags = SHARP_TIP | SHARP_BLADE
+	attack_verb = list("pokes", "stabs", "impales","attacks","flicks","jabs")
+	var/stance = "high"
+	var/cooldown = 0 //cooldown for swapping stances
+
+/obj/item/weapon/rapier/proc/set_low_stance()
+	desc = "It is currently held low. In this stance you can parry attacks. Allez !"
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/weaponry.dm]: space

**Comment by RealestEstate** [comment]
@unid15 what kind of sprite differences did you have in mind? currently the different stances do have a difference in sprites as one is angled up and the other is angled down. Was there something else you had in mind?

File: code/game/objects/items/weapons/weaponry.dm (hunk 3 above)
Comment (RealestEstate) [reply code/game/objects/items/weapons/weaponry.dm]: like between "allez" and "!"?

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (RealestEstate) [reply code/game/objects/items/weapons/weaponry.dm]: I only want it to display the stance switch to the user though

Comment (DamianX) [reply code/game/objects/items/weapons/weaponry.dm]: You can use `to_chat` or `show_message`. The latter can let you take into account the user being blind.

**Comment by unid15** [comment]
Oh, there actually are sprite differences. Could've sworn there weren't, for some reason. Nevermind that then!

**Comment by SGuivenne** [comment]
we need fencer apparel, a higher bleed chance, and a near 0 (if not 0) chance of limb removal if youre going this route.  remember fencing swords are thrusted, not slashed like a katana.

File: code/game/objects/items/weapons/weaponry.dm (hunk 3 above)
Comment (DamianX) [reply code/game/objects/items/weapons/weaponry.dm]: yes

**Comment by DamianX** [comment]
when

**Comment by RealestEstate** [comment]
@DamianX won't be for a while, I'm gonna take a break on this.
in the meantime [DNM]

**Comment by DamianX** [comment]
Shout when you want this reopened

**Pull request closed without merging**